Source/WebCore/inspector/agents/page/PageHeapAgent.cpp

```diff
@@ -58,6 +62,35 @@ Inspector::Protocol::ErrorStringOr<void> PageHeapAgent::disable()
     return WebHeapAgent::disable();
 }
 
+String PageHeapAgent::heapSnapshotBuilderOverrideClassName(JSC::HeapSnapshotBuilder& builder, JSC::JSCell* cell, const String& currentClassName)
+{
+    if (currentClassName == "HTMLElement") {
+        if (auto* jsElement = jsDynamicCast<JSElement*>(cell)) {
+            Ref element = jsElement->wrapped();
+            if (element->isDefinedCustomElement()) {
+                if (RefPtr customElementRegistry = element->customElementRegistry()) {
+                    if (RefPtr customElementInterface = customElementRegistry->findInterface(element)) {
+                        if (JSC::JSObject* customElementConstructorObject = customElementInterface->constructor()) {
```

Comment:
This is a roundabout way to get to the constructor given we already have `JSCell*` but okay.

Reply:
i got this by working backwards from what was provided to `CustomElementRegistry.prototype.define`.  happy to do something else if you think there's a more efficient approach :)

Reply:
Since we already have `JSCell`, we can just get the prototype then its constructor but this code is fine as is.